### DISCUSSION TITLE
About discovery and communication issues with browser and runtime components

### DISCUSSION THREAD

**Discussion by Lafatalite** [pyri-project/pyri-core#13] [Q&A]
I have recently been developing a runtime component. Using the RobotRaconteur framework, I completed the construction of a runtime node without using pyri's method. I can call this runtime component with a RobotRaconteur test node, but I cannot connect to this device in the browser. I also browsed plugin_development.md but did not understand the usage of the files mentioned in the markdown. Thus, I built a node using the RobotRaconteur framework. However, I am now encountering issues accessing it through the browser. Do I need to follow the hints in plugin_development.md to develop the runtime node? And how can I discover a custom device in the browser component? I have defined a robdef file and implemented it in the runtime component. Can you provide me with a simple example of acquiring a runtime object in the browser and calling a function or accessing a variable? These are my files.
[demostartor.zip](https://github.com/user-attachments/files/15510890/demostartor.zip)

**Comment by johnwason** [comment]
Did you figure this out?
